### PR TITLE
Add a link explaning the guard directive used in the code samples

### PR DESCRIPTION
**Pull request by gilberto-torrezan**

### CODE REVIEW

File: articles/ds/components/dialog/index.asciidoc [hunk 1]
```diff
@@ -43,6 +43,9 @@ include::{root}/frontend/demo/component/dialog/dialog-basic.ts[render, tags=*, i
 ----
 --
 
+[IMPORTANT]
+The typescript examples in this page use an inline `.renderer` function, which is encapsulated by the `guard` directive in Lit. This is done for performance reasons. For more details on the `guard` directive, please refer to the https://lit.dev/docs/templates/directives/#guard[official documentation].
```
Comment (tomivirkki) [inline articles/ds/components/dialog/index.asciidoc:47]: The content of the asciidoc should be relevant to the Design System and ideally language independent. Using Lit guards is a technical detail in some examples so this notice would work better as a code comment next to where the `guard` is used.

In future, we'll replace the `guard` directives in the examples with something similar to https://github.com/web-padawan/lit-vaadin-helpers

Comment (magi42) [inline articles/ds/components/dialog/index.asciidoc:47]:
```suggestion
The TypeScript examples in this article use an inline `.renderer` function, which is encapsulated by the `guard` directive in Lit.
The examples do so for performance reasons.
For more details on the `guard` directive, refer to the https://lit.dev/docs/templates/directives/#guard[official Lit documentation].
```

**Review by magi42 — APPROVED** [review]

File: articles/ds/components/dialog/index.asciidoc (hunk 1 above)
Comment (magi42) [reply articles/ds/components/dialog/index.asciidoc]: You can have language-specific notes also in example blocks for each language. Shouldn't use admonitions there though.

I made some language fixes, but maybe this issue should be solved.

Comment (gilberto-torrezan) [reply articles/ds/components/dialog/index.asciidoc]: How do I add notes to the example blocks? I seems to be just linking to an external file.

Comment (magi42) [reply articles/ds/components/dialog/index.asciidoc]: You can have description text above the `[source, Java]` line (leave one empty line between the text). This works inside `[.example]` boxes. Each language (group) can have such description. See for example the code example in https://vaadin.com/docs/latest/flow/application/routing-navigation

**Comment by jouni** [comment]
Oh, damn. I didn't notice this targeted the v20 branch, and I rebased it on latest 🤦

**Comment by jouni** [comment]
I’ll make a new branch, sorry for messing up this one.

**Comment by jouni** [comment]
Pushed to `latest` in https://github.com/vaadin/docs/commit/32ef65648ce0666fc8d95271c56b3bd76a06410e

**Comment by jouni** [comment]
Not sure if it’s relevant to cherry-pick to v20, as that branch will only be around for a month more or so.

**Pull request closed without merging**